spring-cloud-generator/scripts/generate-steps.sh

```diff
@@ -0,0 +1,195 @@
+  buildozer 'delete' WORKSPACE:%maven_install
+  # Add custom maven_install rules
+  perl -pi -e "s{(^_gapic_generator_java_version[^\n]*)}{\$1\n$(cat ../scripts/resources/googleapis_modification_string.txt)}" WORKSPACE
+  # In repository_rules.bzl, add switch for new spring rule
+  JAVA_SPRING_SWITCH="    rules[\\\"java_gapic_spring_library\\\"] = _switch(\n        java and grpc and gapic,\n        \\\"\@spring_cloud_generator\/\/:java_gapic_spring.bzl\\\",\n    )"
+  perl -0777 -pi -e "s/(rules\[\"java_gapic_library\"\] \= _switch\((.*?)\))/\$1\n$JAVA_SPRING_SWITCH/s" repository_rules.bzl
+}
+
+# For individual library, checkout versioned protos folder and set up bazel rules
+#
+# $1 - googleapis commitish
+# $2 - googleapis folder (e.g. google/cloud/accessapproval/v1)
+function prepare_bazel_build(){
+  GOOGLEAPIS_COMMITISH=$1
+  GOOGLEAPIS_FOLDER=$2
+
+  # If googleapis folder does not exist, exit
+  if [ ! -d "$(pwd)/${GOOGLEAPIS_FOLDER}" ]
+  then
+    echo "Directory $(pwd)/${GOOGLEAPIS_FOLDER} does not exist."
+    exit
+  fi
+
+  git checkout ${GOOGLEAPIS_COMMITISH} -- ${GOOGLEAPIS_FOLDER}
+
+  # Modify BUILD.bazel file for library
+  # Additional rule to load
+  SPRING_RULE_NAME="    \\\"java_gapic_spring_library\\\","
+  perl -0777 -pi -e "s/(load\((.*?)\"java_gapic_library\",)/\$1\n${SPRING_RULE_NAME}/s" ${GOOGLEAPIS_FOLDER}/BUILD.bazel
+
+  # Add java_gapic_spring_library rule, with attrs copied from corresponding java_gapic_library rule
+  GAPIC_RULE_NAME="$(buildozer 'print name' ${GOOGLEAPIS_FOLDER}/BUILD.bazel:%java_gapic_library)"
+  SPRING_RULE_NAME="${GAPIC_RULE_NAME}_spring"
+  GAPIC_RULE_FULL="$(buildozer 'print rule' ${GOOGLEAPIS_FOLDER}/BUILD.bazel:%java_gapic_library)"
+
+  buildozer "new java_gapic_spring_library $SPRING_RULE_NAME" ${GOOGLEAPIS_FOLDER}/BUILD.bazel:__pkg__
+
+  # Copy attributes from java_gapic_library rule
+  attrs_array=("srcs" "grpc_service_config" "gapic_yaml" "service_yaml" "transport")
+  for attribute in "${attrs_array[@]}"
+    do
+      echo "$attribute"
+      if [[ $GAPIC_RULE_FULL = *"$attribute"* ]] ; then
+              buildozer "copy $attribute $GAPIC_RULE_NAME" ${GOOGLEAPIS_FOLDER}/BUILD.bazel:$SPRING_RULE_NAME
+          else
+              echo "attribute $attribute not found in java_gapic_library rule, skipping"
+          fi
+    done
+}
+
+# Fetch all `*java_gapic_spring` build rules and build them at once
+function bazel_build_all(){
+  bazelisk query "attr(name, '.*java_gapic_spring', //...)" \
+    | xargs bazelisk build --tool_java_language_version=17 --tool_java_runtime_version=remotejdk_17 2>&1 \
+    | tee tmp-output || save_error_info "bazel_build"
+}
+
+# Add module to aggregator POM in the appropriate location, with check for existence.
+#
+# $1 - path-to-pom-file (e.g. path/to/pom.xml)
+# $2 - starter-artifact-id (e.g. google-cloud-accessapproval-spring-starter)
+function add_module_to_pom () {
+  xmllint --debug --nsclean --xpath  "//*[local-name()='module']/text()" $1 \
```

Comment:
nit: for consistency among functions, we can declare a variable containing `$1` like others do (`path_to_pom=$1`)